### PR TITLE
Feature/GitHub actions

### PR DESCRIPTION
**Pull request by warbhe-amit**
adding github actions for unit test cases

### CODE REVIEW

**Review by michaelcfine — APPROVED** [review]
I didn't go through every line of every file, but looks okay. One comment: The test for the dot config file should not check against the standard/default/sample config file... the whole point is that users can change their config file to contain whatever apis they want (and change the other settings, too). So it probably only makes sense to test that the file contents are json or something like that.

**Review by michaelcfine — DISMISSED** [review]
I'm not familiar with github actions, but if this all works, looks good to me. We do want to make the unit test check mandatory, though.

**Review by kschrief — APPROVED** [review]

**Comment by warbhe-amit** [comment]
@michaelcfine Please review the dotConfig test file
We'll merge this branch once PR 49 gets merged into main.

File: server/test/suite/dotConfig.test.mjs
Comment (michaelcfine) [inline server/test/suite/dotConfig.test.mjs:26]: It seems like you are testing that the dotConfig object within the dotConfig module is a valid object. Not sure that's valuable. The code that loads the dotConfig file already has a try-catch when it tries to parse the .mf.config.json file.

So... what's the ultimate point of this test? Do we really need it?

Comment (warbhe-amit) [reply server/test/suite/dotConfig.test.mjs:26]: Taken care

File: .github/workflows/npmPublish.yml
Comment (kschrief) [inline .github/workflows/npmPublish.yml:8]: This indicates that the job will run on every release. We want the job to be strictly manual for now.

**Review by kschrief — CHANGES_REQUESTED** [review]

File: .github/workflows/npmPublish.yml
Comment (warbhe-amit) [reply .github/workflows/npmPublish.yml]: added manual configuration on npm publish

**Review by kschrief — APPROVED** [review]

**Review by kschrief — DISMISSED** [review]

**Review by michaelcfine — DISMISSED** [review]
👍

**Review by kschrief — DISMISSED** [review]

**Comment by warbhe-amit** [comment]
code coverage is 78.4%

**Review by kschrief — APPROVED** [review]

**Review by dhillomk — APPROVED** [review]

**Pull request merged**
